src/v3/test/integration/authenticator-piv-cac-verification.test.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import mockResponse from '../../src/mocks/response/idp/idx/introspect/identify-with-piv-cac.json';
+import { setup } from './util';
+
+describe('authenticator-piv-cac-verification', () => {
+  it('renders form with idpDisplay as PRIMARY/default', async () => {
+    const { container, findByLabelText } = await setup({ mockResponse });
+    await findByLabelText(/Username/);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders form with idpDisplay as SECONDARY', async () => {
+    const { container, findByLabelText } = await setup({
+      mockResponse, widgetOptions: { idpDisplay: 'SECONDARY' },
+    });
+    await findByLabelText(/Username/);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders form with custom PIV/CAC Button label and class', async () => {
+    const { container, findByLabelText } = await setup({
+      mockResponse,
+      widgetOptions: { piv: { text: 'My custom PIV Button Label', className: 'custom-piv-class' } },
+    });
+    await findByLabelText(/Username/);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render PIV/CAC view when clicking PIV button', async () => {
+    const {
+      container, user, findByRole, findByText,
+    } = await setup({ mockResponse });
+
+    const pivButton = await findByText('Sign in with PIV / CAC card', { selector: 'button' });
+
+    await user.click(pivButton);
```

Comment:
Yep, we totally can, I didnt initially because we wont be able to use the util function, but I will add the custom assertion here for the request. 🙏

Reply:
Actually, a correction to this, we can't assert on the request because we are using a v2 util function that loads a hidden form on the page (with the action value being the MTLS endpoint) and submits it. There doesn't seem to be a way for me to spy this to verify. 

But I am adding a unit test to the PIVButton component to assert it triggers the click as expected.

Reply:
Ah okay, perhaps this is something that is coverable in the testcafe suites since those have network spies.